cpp/src/arrow/dataset/dataset.h

```diff
@@ -53,6 +53,23 @@ class ARROW_DS_EXPORT DataFragment {
   virtual ~DataFragment() = default;
 };
 
+/// \brief A trivial DataFragment that yields ScanTask out of a fixed set of
+/// RecordBatch.
+class ARROW_DS_EXPORT SimpleDataFragment : public DataFragment {
+ public:
+  explicit SimpleDataFragment(std::vector<std::shared_ptr<RecordBatch>> record_batches);
+
+  Status Scan(std::shared_ptr<ScanContext> scan_context,
+                  std::unique_ptr<ScanTaskIterator>* out) override;
+
+  bool splittable() const override { return false; }
```

Comment:
I'm not sure about this interface, I think we should delete until we make use of it, YAGNI.

Reply:
sounds fine to me; sub-file data fragments sounds like something we'd want to handle at the DataSource level anyhow